Scripts/Process/Component/GameObjectSourceTargetProcessor.cs

```diff
@@ -0,0 +1,187 @@
+        {
+            if (sources.Count == 0)
+            {
+                sources.Add(source);
+            }
+            else
+            {
+                sources[sources.GetWrappedAndClampedIndex(index)] = source;
+            }
+        }
+
+        /// <summary>
+        /// Sets the given source at the current sources index.
+        /// </summary>
+        /// <param name="source">The source to set.</param>
+        public virtual void SetSourceAtCurrentIndex(GameObject source)
+        {
+            SetSource(CurrentSourcesIndex, source);
+        }
+
+        /// <summary>
+        /// Clears the sources collection.
+        /// </summary>
+        public virtual void ClearSources()
+        {
+            sources.Clear();
+        }
+
+        /// <summary>
+        /// Adds the given target to the targets collection.
+        /// </summary>
+        /// <param name="target">The target to add.</param>
+        public virtual void AddTarget(GameObject target)
+        {
+            targets.Add(target);
+        }
+
+        /// <summary>
+        /// Removes the given target from the targets collection.
+        /// </summary>
+        /// <param name="target">The target to remove.</param>
+        public virtual void RemoveTarget(GameObject target)
+        {
+            targets.Remove(target);
+        }
+
+        /// <summary>
+        /// Sets the given target at the given index of the targets collection.
+        /// </summary>
+        /// <param name="index">The index to set the target at.</param>
+        /// <param name="target">The target to set.</param>
+        public virtual void SetTarget(int index, GameObject target)
+        {
+            if (targets.Count == 0)
+            {
+                targets.Add(target);
+            }
+            else
+            {
+                targets[targets.GetWrappedAndClampedIndex(index)] = target;
+            }
+        }
+
+        /// <summary>
+        /// Sets the given target at the current targets index.
+        /// </summary>
+        /// <param name="target">The target to set.</param>
+        public virtual void SetTargetAtCurrentIndex(GameObject target)
+        {
+            SetTarget(CurrentTargetsIndex, target);
+        }
+
+        /// <summary>
+        /// Clears the targets collection.
+        /// </summary>
+        public virtual void ClearTargets()
+        {
+            targets.Clear();
+        }
+
+        /// <inheritdoc />
+        protected override void SetCurrentIndices(int sourceIndex, int targetIndex)
+        {
+            CurrentSourcesIndex = sourceIndex;
+            CurrentTargetsIndex = targetIndex;
+        }
+
+        /// <inheritdoc />
+        protected override bool IsSourceValid(GameObject source)
+        {
+            return (base.IsSourceValid(source) && sourceValidity.Accepts(source));
+        }
+
+        /// <inheritdoc />
+        protected override bool IsTargetValid(GameObject target)
```

Comment:
Same as above: Does it make sense for the superclass to have this validity stuff?